find_recipes.py

```diff
@@ -0,0 +1,58 @@
+import pickle
+import recipe
+from collections import defaultdict
+
+
+with open('raw_data/pickle_files/ingredient_to_recipes_dict.pickle', 'rb') as f:
+    ingredient_to_recipes = pickle.load(f)
+
+with open('raw_data/pickle_files/recipe_toingredients_toquantities.pickle', 'rb') as f:
+    recipes_to_quantities = pickle.load(f)
+
+def check_recipe_validity(recipe):
+    for key in recipes_to_quantities[recipe.id]:
+        totals = recipes_to_quantities[recipe.id][key]
+        ## Occurs if recipe does not specify quantity for an ingredient, for this arbitrary measure we assume any amount the user has is sufficient, provided they have an ammount
+        if totals == [0, 0, 0, 0]:
+            if user_ingredients[key] == [0, 0, 0, 0]:
+                return False
+        ## Check if the user has a sufficient amount of the provided ingredient
+        for x in range(4):
+            if user_ingredients[key][x] < totals[x]:
```

Comment:
Will need to add in actual value for user_ingredients when UI is more developed, for now just imagine it as a dictionary with the keys being ingredients and the values being [0, 0, 0, 0] to represent quantities.